data-node/src/main/java/org/graylog/datanode/DirectoryReadableValidator.java

```diff
@@ -27,14 +30,21 @@
  * or permissions problem. This validator provides more specific exception with detailed message.
  */
 public class DirectoryReadableValidator extends DirectoryPathReadableValidator {
+    private static final Logger LOG = LoggerFactory.getLogger(DirectoryReadableValidator.class);
+
     @Override
     public void validate(String name, Path value) throws ValidationException {
         if (value == null) {
             return;
         }
 
         if (!Files.exists(value)) {
-            throw new ValidationException("Cannot read from directory " + name + " at path " + value + ". Directory doesn't exist. Please create the directory.");
+            try {
+                Files.createDirectories(value);
```

Comment:
I am not sure if it makes sense to **create** directories in a **read** validator. So far I have used this validator in places that explicitly count with the read-only situation and we also document such usages as read-only dirs.

Reply:
I roll back that change and created a discussion topic to address this in. different PR